students/eowyn/session09/oo-mailroom/test_oomailroom.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+import oomailroom as mr
+import os.path
+
+
+def test_donor_init():
+    name = "Ada Lovelace"
+    amt = 345
+    d = mr.Donor(name)
+    d.add_donation(amt)
+    assert d.name == name
+    assert d.donations == [345]
+
+def test_name():
+    name = "Ada Lovelace"
+    amt = 345
+    d = mr.Donor(name)
+    d.add_donation(amt)
+    assert d.name == name
+
+
+def test_create_transactions():
+    name = "Ada Lovelace"
+    amt = 345
+    d = mr.Donor(name)
+    d.add_donation(amt)
+    t = mr.Transactions()
+    t.add_donor(name, amt)
+    assert t.get_donor(name).name == name
+
+
+def test_add_donors_to_transactions():
+    name = "Ada Lovelace"
+    amt = 345
+    d = mr.Donor(name)
+    d.add_donation(amt)
+    name2 = "Marge Simpson"
+    amt2 = 500
+    d2 = mr.Donor(name2)
+    d2.add_donation(amt2)
+    t = mr.Transactions()
+    t.add_donor(name, amt)
+    t.add_donor(name2, amt2)
+    assert t.get_donor(name).name == name
+    assert t.get_donor(name2).name == name2
+    assert t.get_donor(name).most_recent_donation == amt
+    assert t.get_donor(name2).most_recent_donation == amt2
```

Comment:
good tests, but mingling things a bit -- you're testing BOTH get_donor AND Donor.most_recent_transaction.

better to test those separately.